MantidQt/MantidWidgets/src/InstrumentView/Shape2DCollection.cpp

```diff
@@ -685,7 +685,9 @@ void Shape2DCollection::loadFromTableWorkspace(
   for (size_t i = 0; i < ws->rowCount(); ++i) {
     const auto params = col[i];
     auto shape = Shape2D::loadFromProject(params);
-    m_shapes.append(shape);
+    if (shape != nullptr) {
```

Comment:
Ahh from the `auto` type I assumed this would be a pointer type. 
If it could be integer as implied with point 1. I agree this check should be against a `nullptr` type.